lstests/core/service.py

```diff
@@ -0,0 +1,1517 @@
+        self.cla = CommandLineArguments()
+        # configure service
+        self.addr  = helper.override(DEFAULT_ADDR,  addr,  self.cla.addr)
+        self.port  = helper.override(DEFAULT_PORT,  port,  self.cla.port)
+        self.wdir  = helper.override(DEFAULT_WDIR,  wdir,  self.cla.wdir)
+        self.file  = helper.override(DEFAULT_FILE,  file,  self.cla.file)
+        self.gene  = helper.override(DEFAULT_GENE,  gene,  self.cla.gene)
+        self.start = helper.override(DEFAULT_START, start, self.cla.start)
+        self.kill  = helper.override(DEFAULT_KILL,  kill,  self.cla.kill)
+        # change working dir
+        os.chdir(self.wdir)
+        # register logger
+        self.register_logger(logger)
+        # connect
+        self.connect()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exception_type, exception_value, exception_traceback):
+        """Flush all buffered logging information in case Service crashes"""
+        self.flush()
+
+    def register_logger(self, logger):
+        self.logger = logger
+        for w in self.logger.writers:
+            # override screen writer settings with command-line arguments
+            if isinstance(w, ScreenWriter):
+                self.threshold = w.threshold = LogLevel(helper.override(w.threshold, self.cla.threshold))
+                self.buffered = w.buffered = helper.override(DEFAULT_BUFFERED, w.buffered, self.cla.buffered)
+                self.monochrome = w.monochrome = helper.override(DEFAULT_MONOCHROME, w.monochrome, self.cla.monochrome)
+                if self.cla.hide_all:
+                      w.show_clock_time = w.show_elapsed_time = w.show_filename = w.show_lineno = w.show_function = \
+                      w.show_thread = w.show_log_level = False
+                else:
+                    w.show_clock_time = helper.override(DEFAULT_SHOW_CLOCK_TIME, w.show_clock_time, self.cla.show_clock_time)
+                    w.show_elapsed_time = helper.override(DEFAULT_SHOW_ELAPSED_TIME, w.show_elapsed_time, self.cla.show_elapsed_time)
+                    w.show_filename = helper.override(DEFAULT_SHOW_FILENAME, w.show_filename, self.cla.show_filename)
+                    w.show_lineno = helper.override(DEFAULT_SHOW_LINENO, w.show_lineno, self.cla.show_lineno)
+                    w.show_function = helper.override(DEFAULT_SHOW_FUNCTION, w.show_function, self.cla.show_function)
+                    w.show_thread = helper.override(DEFAULT_SHOW_THREAD, w.show_thread, self.cla.show_thread)
+                    w.show_log_level = helper.override(DEFAULT_SHOW_LOG_LEVEL, w.show_log_level, self.cla.show_log_level)
+            # empty log files
+            elif isinstance(w, FileWriter):
+                if not self.cla.dont_rename and self.cla.cluster_id is not None:
+                    bas, ext = os.path.splitext(w.filename)
+                    w.filename = f"{bas}-{self.cla.cluster_id}{ext}"
+                with open(w.filename, "w"):
```

Comment:
Thank you for the comment. The logic here is: after the initialization of the logger/writers, a `Service` object will have a reference to, and thus the control over, the logger. The `Service` object may overwrite the logger settings according to the command-line arguments. In particular, the cluster ID is usually provided by the user at the command line and cannot be known in advance at the stage of logger/writers' initialization.

By the way, the reason to provide log file renaming functionality is to allow for a parallel run of one test script on multiple clusters. Consider in the script the log file name is `test.log`. But in practice, we may want to run two instances simultaneously, on clusters # 0 and # 1 respectively. Then the log files will be renamed as `test-0.log` and `test-1.log` to avoid the conflict.